lib/ConsumerImpl.cc

```diff
@@ -331,6 +371,13 @@ Optional<SharedBuffer> ConsumerImpl::processMessageChunk(const SharedBuffer& pay
                                                  << payload.readableBytes() << " bytes");
 
     Lock lock(chunkProcessMutex_);
+
+    // Lazy task scheduling to expire incomplete chunk message
+    if (!checkExpiredChunkedTimer_) {
+        checkExpiredChunkedTimer_ = executor_->createDeadlineTimer();
+        triggerCheckExpiredChunkedTimer();
+    }
```

Comment:
Just create the timer in the constructor. `createDeadlineTimer` won't start any timer, it only creates the object. Otherwise, we should always check it's not a nullptr when accessing it.

Reply:
The `checkExpiredChunkedTimer_ ` should always be not null in `triggerCheckExpiredChunkedTimer `; outside this method, we have always checked the null for it.
If we go this way, we should always get the state of checkExpiredChunkedTimer_ here to check if the timer has been started to implement lazy scheduling. I haven't found a more elegant way to get the state. I think the current approach should be better.

Reply:
We should add a boolean variable `expireChunkMessageTaskScheduled` like Java client does. You can also use something like `Optional` but since `deadline_timer` is not copyable you have to use the move constructor.

BTW, I am going to remove unnecessary `std::shared_ptr` usages in C++ client. It should not be used everywhere. When writing C++, the best practice is to make the lifetime clear like writing Rust.

Reply:
https://godbolt.org/z/eezdxMGnc @Demogorgon314

Reply:
You're right. Agree to make the lifetime clearer.